CONTRIBUTING.md

```diff
@@ -0,0 +1,193 @@
+        -   [Do your work on a branch in your fork](#do-your-work-on-a-branch-in-your-fork).
+    -   2.B: You have made changes.
+        -   2.B.1: You are working on the main branch.
+            -   2.B.1.A: You made commits.
+                -   TODO
+            -   2.B.1.B: You haven't made commits.
+                -   [Stash your changes](#stash-your-changes).
+                -   [Create a branch](#create-a-branch).
+                -   [Do your work on a branch in your fork](#do-your-work-on-a-branch-in-your-fork).
+        -   2.B.2: You are working on a different branch.
+            -   2.B.2.A: You have made commits.
+                -   [Do your work on a branch in your fork](#do-your-work-on-a-branch-in-your-fork).
+            -   2.B.2.B: You haven't made commits.
+                -   [Commit your changes](#commit-changes).
+                -   [Do your work on a branch in your fork](#do-your-work-on-a-branch-in-your-fork).
+
+## Stash your changes
+
+To "stash" your changes (i.e. squirrel them away for later use), issue the
+following command:
+
+```sh
+git stash
+```
+
+## Create a branch
+
+To create a branch and switch to it in one command, issue the following
+command:
+
+```sh
+git checkout -b your_branch_name_here
+```
+
+At Numberscope we use `snake_case` for branch names.
+
+## Push a branch
+
+To create a branch in a remote Git repository and push your changes to that
+branch, issue the following command:
+
+```sh
+git push -u name_of_remote name_of_branch
+```
+
+The name of the remote is most likely `origin` if you are pushing to a fork of
+one of the Numberscope repositories.
+
+## Create a fork
+
+Note: This is a GitHub operation, not a Git operation.
+
+1. Go to the page of the repository you want to clone. For instance, if you
+   want to clone the numberscope/frontscope repository, go to
+   https://github.com/numberscope/frontscope.
+2. In the upper right corner of the page (as of this writing) there should be
+   a button that says "Fork". Click that button.
+3. Follow the instructions that GitHub provides. Make your GitHub account the
+   owner of the fork.
+
+## Clone a repo
+
+To clone a Git repository to your computer, issue the following command:
+
+```sh
+git clone https://github.com/some_user_or_org/some_repo.git
+```
+
+GitHub allows you to clone the repo a few different ways:
+
+1. via HTTPS (easiest method, doesn't require setup, has limited
+   functionality)
+2. [via SSH](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account)
+   (hardest method, requires setup, but ultimately very convenient)
+3. [via the GitHub CLI](https://cli.github.com/) (medium difficulty, requires
+   some setup)
+
+## Add a remote
+
+To add a remote version of your repository, issue the following command:
+
+```sh
+git remote add name_of_remote https://someurl.com/somerepo.git
+```
+
+## Do your work on a branch in your fork
+
```

Comment:
This title sounds like a command rather than a section title.  Maybe "How to contribute code:  add, commit and push"  Then you could add a sentence that you are working on your branch in your fork.